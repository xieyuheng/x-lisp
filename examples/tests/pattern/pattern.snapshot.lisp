;; var

(@pattern x)
(@pattern _)

;; tael

(@pattern [a b c :x x :y y])
(@pattern [a b c :x x :y y :z z])

;; literal

(@pattern 1)
(@pattern "a")
(@pattern #void)
(@pattern #null)
(@pattern 'a)
(@pattern '(a b c))

;; cons and cons*

(@pattern (cons head tail))
(@pattern (cons* first second tail))

;; escape

(@pattern (cons* first (@escape (iadd 1 1)) tail))
(@pattern (cons* first (@escape (iadd 1 2)) tail))
(@pattern (@escape null))

;; quasiquote

(@pattern `(,first ,(@escape (iadd 1 1)) 3))
(@pattern `(,first ,(@escape (iadd 1 2)) 3))

;; occur many times

(@pattern [x x])
