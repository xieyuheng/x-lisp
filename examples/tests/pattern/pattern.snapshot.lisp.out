(@pattern x)
(@pattern _)
(@pattern [a b c :x x :y y])
(@pattern [a b c :x x :y y :z z])
(@pattern 1)
(@pattern "a")
(@pattern #void)
(@pattern #null)
(@pattern 'a)
(@pattern (@escape ['a 'b 'c]))
(@pattern (cons* head tail))
(@pattern
  (cons* first second tail))
(@pattern (cons* first 2 tail))
(@pattern (cons* first 3 tail))
(@pattern #null)
(@pattern [first 2 3])
(@pattern [first 3 3])
(@pattern [x x])
